src/components/context/data/processors/jcampProcessor.ts

```diff
@@ -0,0 +1,81 @@
+import { PartialFileList } from 'filelist-utils';
+import { convert } from 'jcampconverter';
+import { v4 } from 'uuid';
+
+import { DataState, MeasurementKind } from '../DataState';
+import { irSpectrumEnhancer } from '../enhancers/irSpectrumEnhancer';
+
+const enhancers: Record<string, typeof irSpectrumEnhancer> = {
+  ir: irSpectrumEnhancer,
+};
+
+export async function jcampProcessor(
+  fileList: PartialFileList,
+  dataState: DataState,
+) {
+  for (const file of fileList) {
+    if (/(.jdx|.dx)$/i.exec(file.name)) {
+      const parsed = convert(await file.text(), { keepRecordsRegExp: /.*/ });
+      for (const measurement of parsed.flatten) {
+        let kind: MeasurementKind | '' = '';
+        if (measurement?.dataType?.match(/(infrared|ir)/i)) {
+          kind = 'ir';
+        }
+        if (measurement?.dataType?.match(/raman/i)) {
+          kind = 'raman';
+        }
+        if (measurement?.dataType?.match(/uv/i)) {
+          kind = 'uv';
+        }
+        if (kind) {
+          dataState.measurements[kind].entries.push({
+            id: v4(),
+            meta: measurement.meta,
+            filename: file.name,
+            path: file.webkitRelativePath,
+            //@ts-expect-error measurement.info should be ok
+            info: measurement.info,
+            title: measurement.title,
+            data: normalizeSpectra(measurement.spectra, kind),
+          });
+        }
+      }
+    }
+  }
+}
+
+function normalizeSpectra(spectra: any, kind: string) {
```

Comment:
spectrum type in jcampconverter
```
/**
 * @typedef { Record<string, any> } Spectrum
 * @property {Record<string, number[]>} [data]
 * @property {number} [firstX]
 * @property {number} [lastX]
 * @property {number} [deltaX]
 * @property {number} [yFactor]
 * @property {number} [xFactor]
 * @property {number} [nbPoints]
 */
```

Reply:
We will update this when jcampconverter will be migrate to typescript. Thanks !